test/e2e/framework/fips.go

```diff
@@ -0,0 +1,89 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	fipsNamespace     = "kube-system"
+	fipsConfigMapName = "cluster-config-v1"
+)
+
+func FindFipsConfig(cluster ClusterIndex) bool {
+	configMap, err := KubeClients[cluster].CoreV1().ConfigMaps(fipsNamespace).Get(context.TODO(), fipsConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			return false
+		}
+	}
+
+	if strings.Contains(configMap.Data["install-config"], "fips: true") {
+		return true
+	}
+
+	return false
+}
+
+func (f *Framework) FindFipsEnabledCluster() int {
+	for idx := range TestContext.ClusterIDs {
+		fipsEnabled := FindFipsConfig(ClusterIndex(idx))
+		if fipsEnabled {
+			return idx
+		}
+	}
+
+	return -1
+}
+
+func (f *Framework) TestGatewayNodeFipsMode(cluster ClusterIndex, gwPod string) error {
+	ctx := context.TODO()
+	cmd := []string{"sh", "-c", "ipsec pluto --selftest"}
+
+	stdOut, stdErr, err := f.ExecWithOptions(ctx, &ExecOptions{
+		Command:       cmd,
+		Namespace:     TestContext.SubmarinerNamespace,
+		PodName:       gwPod,
+		ContainerName: SubmarinerGateway,
+		CaptureStdout: true,
+		CaptureStderr: true,
+	}, cluster)
+	Expect(err).To(BeNil())
+
+	if stdOut == "" && stdErr == "" {
+		return fmt.Errorf("no output has been detected from the %q cmd", cmd)
+	}
```

Comment:
On L73 it asserts no error but on L76 it returns an error. We should be consistent. I think since this function is advertised to perform test(s), it should assert success or failure in all cases and not return an error. 



```suggestion
	if stdOut == "" && stdErr == "" {
	         Fail(fmt.Sprintf("No output received from command %q", cmd))
	}
```

Reply:
Done